activerecord/lib/active_record/reflection.rb

```diff
@@ -416,6 +416,9 @@ def mapping
     # Active Record class.
     class AssociationReflection < MacroReflection #:nodoc:
       def compute_class(name)
+        if polymorphic?
+          raise NameError, "Polymorphic association does not support to compute class."
```

Comment:
Hmm.. I see the advantage of leaning on the existing code, but I think making this a NameError is still a bit confusing: normally a NameError here is the fault of the data (a bad/unknown class name), whereas in this case it's the fault of the calling code (breaking a contract by calling a method when it shouldn't). Better to keep them separate, I think -- and then pull in the conditional from #31877 to ensure we conform to that contract.

(Otherwise some other code might also rescue NameError, and do something else -- and then they would miss out on the warning that they're misusing the API.)

Reply:
Your requested changes are

* Do not use `NameError` (use `ArgumentError` instead, or another?)
* Do not attempt to automatically find the inverse of a polymorphic not to attempt to call `klass` of polymorphic

Right?